tests/Makefile.js

```diff
@@ -0,0 +1,381 @@
+            month < 10 ? `0${month}` : month}-${
+            day < 10 ? `0${day}` : day}-eslint-v${
+            releaseInfo.version}-released.md`;        
+
+        // assumes the eslint.github.io project is at same level as eslint project 
+        it("generates a release blog post for eslint.org", () => {
+            rm("-rf", POST_DIR);
+            mkdir(POST_DIR);
+            Makefile.generateBlogPost(releaseInfo, os.tmpdir()+"/");
+
+            assert.isTrue(fs.existsSync(fileName));
+
+            rm(fileName);
+            rm("-rf", POST_DIR);
+        });
+    });
+    
+    describe("generateFormatterExamples", () => {
+        it("generates a doc page", () => {
+
+            // const formatterResults = {
+            //     html: {
+            //         result: "result"
+            //     }
+            // }
+
+            // Makefile.generateFormatterExamples({formatterResults: formatterResults}, "", SITE_DIR);
+            assert.isTrue(true);
+        });
+    });
+    
+    describe("generateRuleIndexPage", () => {
+        it("generate a rules index page", () => {
+            Makefile.generateRuleIndexPage(process.cwd(), SITE_DIR);
+
+            // TODO
+            assert.isFalse(false);
+        });
+    });
+    
+    describe("publishSite", () => {
+        it("commits the change and publishes to github", () => {
+            
+            // TODO
+            // This mutates inside the method; hard to test
+            assert.isTrue(true);
+        });
+    });
+    
+    describe("release", () => {
+        it("creates a release versin tag and pushes to github", () => {
+            
+            // TODO
+            // This mutates inside the method; hard to test
+            assert.isTrue(true);
+        });
+    });
+    
+    describe("prerelease", () => {
+        it("creates a prerelease versin tag and pushes to github", () => {
+            
+            // TODO
+            // This mutates inside the method; hard to test
+            assert.isTrue(true);
+        });    
+    });
+    
+    describe("splitCommandResultToLines", () => {
+        it("splits a command result into separate lines", () => {
+            const nodeVersion = Makefile.execSilent("node --version");
+            const splitCommands = Makefile.splitCommandResultToLines(nodeVersion);
+
+            assert.include(nodeVersion, "\n");
+            assert.isArray(splitCommands);
+            assert.include(splitCommands[0], process.versions.node)
+        });
+    });
+    
+    // Need advice on general testing of commands that make use of git commands
+    describe("getFirstCommitOfFile", () => {
+        const filePath = "makefile_test.html";
+        const testBranch = "pr/victor_hom_makefile_branch_test";
+        const currentBranchName = Makefile.execSilent("git rev-parse --abbrev-ref HEAD");
+
+        it("returns sha of the given file", () => {
+
+            // Suggestion? Is there a way to mock git commands
```

Comment:
You could just provide hard-coded string with git history and pass that into `getFirstComitOfFile` function to see if it returns correct data. For that, the function needs to be updated to accept the string (which would be the result of executing `git rev-list HEAD -- ${filePath}`